src/sqlfluff/rules/std.py

```diff
@@ -2827,3 +2827,91 @@ def _eval(self, segment, **kwargs):
+
+    | **Anti-pattern**
+
+    .. code-block:: sql
+
+        select
+            a,
+            *,
+            row_number() over (partition by id order by date) as y,
+            b
+        from x
+
+
+    | **Best practice**
+    |  Order "select" targets in ascending complexity
+
+    .. code-block:: sql
+
+        select
+            *,
+            a,
+            b,
+            row_number() over (partition by id order by date) as y
+        from x
+
+    """
+
+    def _validate(self, i, segment):
+        # Check if we've seen a more complex select target element already
+        if self.seen_element_band[i + 1::] != [None] * len(self.seen_element_band[i + 1::]):
+            # Not quite worked out the fix
+            # earliest_bad_element = next(el for el in self.seen_element_band[i+1::] if el is not None)
+            self.violation_buff.append(
+                LintResult(
+                    anchor=segment,
+                    # fixes=[LintFix('create', earliest_bad_element, segment),
+                    #        LintFix('delete', segment)]
+                )
+            )
+        self.current_element_band = i
+        if not self.seen_element_band[i]:
+            # If it's the first time we've seen this target element add it to the seen_element_band list
+            self.seen_element_band[i] = segment
+
+    def _eval(self, segment, **kwargs):
+
+        self.violation_buff = []
+        # Bands of select targets in order to be enforced
+        select_element_order_preference = (
+            ('wildcard_expression',),
+            ('object_reference', 'literal', 'cast_expression', ('function', 'cast')),
+        )
+        # Track which bands have been seen, with additional None to track 'other' elements
+        # If we find a matching target element, we add the element to the corresponding index
+        self.seen_element_band = [None for i in select_element_order_preference] + [None]
+
+        if segment.type == 'select_clause':
+            select_target_elements = segment.get_children('select_target_element')
+            if not select_target_elements:
+                return None
+
+            for segment in select_target_elements:
+                # The band index of the current segment in select_element_order_preference
+                self.current_element_band = None
+                # Identify target elements that are in select_element_order_preference
+                for i, band in enumerate(select_element_order_preference):
+                    for e in band:
+                        # Identify function type select target
+                        if type(e) == tuple:
+                            if e[0] == 'function':
+                                if segment.get_child('function'):
+                                    if segment.get_child('function').get_child('function_name'):
+                                        if segment.get_child('function').get_child('function_name').raw == e[1]:
```

Comment:
There also has to be a nicer way of doing this other than a try: except

Reply:
does `.recursive_crawl()` help at all rather than successive `.get_child()`?

Reply:
It could but I think I prefer the get_child as it is more specific - I've gone for a try except as while it's a bit anti-establishment it's more readable than my previous approach